src/benchmark.py

```diff
@@ -179,77 +179,74 @@ def test_small_map():
-        \tTime: {tstime}
-        """.format(
-          nnroute=nn_routes,
-          nndist=nn_distance,
-          nntime=nn_time,
-          toroute=to_routes,
-          todist=to_distance,
-          totime=to_time,
-          randroute=rand_routes,
-          randdist=rand_distance,
-          randtime=rand_time,
-          torandroute=to_rand_routes,
-          torandomdist=to_rand_distance,
-          torandomtime=to_rand_time,
-          glsroute=gls_routes,
-          glsdistance=gls_distance,
-          glstime=gls_time,
-          tsroute=ts_routes,
-          tsdistance=ts_distance,
-          tstime=ts_time
-        ))
+  draw_map(coords, config_name, True)
+  draw_solution(coords, nn_routes, config_name + ' nearest neighbour')
+  draw_solution(coords,to_routes, config_name + ' two-opt (NN)')
+  draw_solution(coords, rand_routes, config_name + ' random solution')
+  draw_solution(coords, to_rand_routes, config_name + ' two-opt (random input)')
+  draw_solution(coords, gls_routes, config_name + ' guided local search')
+  draw_solution(coords, ts_routes, config_name + ' tabu search')
+
+
+def test_fleets(model, coords, vehicle_capacity, fleet_ratio, num_locations):
+  """
+  Test different fleet configurations on the same network of locations and demands
+  """
+  for ratio in fleet_ratio:
+    fleet_size = int(num_locations * ratio)
+    model = change_fleet_config(model, fleet_size, [vehicle_capacity]*fleet_size, num_locations)
+    test_all_algorithms(model, coords, '{x}V,{y}L,equal_capacity'.format(x=fleet_size,y=num_locations))
+
+
+def test_fleet_configs_on_maps(fleet_ratios, location_counts, vehicle_capacity):
+  """
+  Runs each algorithm on a set of maps. Tests multiple fleet configurations for each map.
+  Each algorithm runs every combination of fleet size and location count.
+  """
+  for location_count in location_counts:
+    print('testing {i} locations'.format(i=location_count))
+    # initially created using 1 vehicle with enough capacity to hit every location. Modified in test_fleets
+    model, coords = create_model(1000, location_count, 1, [location_count])
+    test_fleets(model, coords, vehicle_capacity, fleet_ratios, location_count)
+
+
+def change_fleet_config(model, num_vehicles, capacities, num_locations):
+  model['num_vehicles'] = num_vehicles
+  model['vehicle_capacities'] = capacities
+  model['demands'] = assign_demand(num_locations, np.sum(capacities), min(capacities))
+  return model
 
 
 def benchmark_suite():
   """
   Run a series of tests on each algorithm and save the results
   """
   print('starting benchmarks...')
-  test_small_map()
+  # this is the temporary code for debugging why or_tools is returning a None solution... hmm
+  # model, coords = create_model(1000, 10, 5, [200]*5)
+  # routes, distance, exec_time = test_guided_local_search(model)
+  
+  # This is the real test, different fleets and locations
+  fleet_ratios = [0.75, 0.5, 0.25, 0.15]
+  location_counts = [100]
```

Comment:
also here, should we put all the counts: 10, 50, 100, 200 here?

Reply:
done